### PR TITLE
Removed extra call addPostFiles method

### PR DESCRIPTION
**Pull request by spolischook**

### CODE REVIEW

File: Goutte/Client.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ protected function doRequest($request)
             $guzzleRequest->setHeader($name, $value);
         }
 
-        if ('POST' == $request->getMethod() || 'PUT' == $request->getMethod()) {
+        if (in_array($request->getMethod(), ['POST', 'PUT']) && $request->getFiles()) {
```
Comment (fabpot) [inline Goutte/Client.php]: in_array is (probably) slower than the previous code

Comment (spolischook) [inline Goutte/Client.php]: let me check

Comment (spolischook) [inline Goutte/Client.php]: You are right!
I've check 1000000 (million) times every variant:
```
== || == 0.24201488494873 seconds
in_array 0.77801990509033 seconds
```
However my variant is more readable

Comment (fabpot) [inline Goutte/Client.php]: Well, readability is subjective, I prefer the old way :)

Comment (spolischook) [inline Goutte/Client.php]: you are owner, so as you wish (:

**Comment by fabpot** [comment]
Thank you @spolischook.

**Pull request closed without merging**